app/Http/Controllers/PaymentController.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\CreatePaymentRequest;
+use App\Models\Transaction;
+use App\Services\PaymentService;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\Request;
+use Illuminate\View\View;
+
+class PaymentController extends Controller
+{
+    public function store(CreatePaymentRequest $request, PaymentService $paymentService): RedirectResponse
+    {
+        $transaction = Transaction::factory()->create(['cents_charged' => $request->amount, 'order_id' => $request->orderId]);
```

Comment:
I wouldn't use the factories in application code and only in the tests/seeders as it requires faker (a dev dependency).